tests/functional/conftest.py

```diff
@@ -298,16 +301,17 @@ def _find_executables(self, name):
         # For Windows append .exe extension to patterns.
         if is_win:
             patterns = [pt + '.exe' for pt in patterns]
-        # For Mac OS X append pattern for .app bundles.
-        if is_darwin:
-            # e.g:  ./dist/name.app/Contents/MacOS/name
-            pt = os.path.join(self._distdir, name + '.app', 'Contents', 'MacOS', name)
-            patterns.append(pt)
         # Apply file patterns.
         for pattern in patterns:
             for prog in glob.glob(pattern):
                 if os.path.isfile(prog):
                     exes.append(prog)
+        # Replace any executable with a Mac OS X .app bundle if it exists.
+        if is_darwin:
+            # e.g:  ./dist/name.app/Contents/MacOS/name
+            pt = os.path.join(self._distdir, name + '.app', 'Contents', 'MacOS', name)
+            if os.path.isfile(pt):
+                exes = [pt]
```

Comment:
This change always returns a *single* executable only. I'm afraid, this will break the intended behavior of this method: It is expected to return a list of all executables a test-case has generated. This is why there is a pattern generated near the top.

If your change is really intended to do so, this should
1. have an explaining comment (I do not understand the reason, even when reading the pull-requests discussion),
2. go to the top of the method, returning the result there and a) avoid the pattern-matching at all, and b) being easy to spot a a special case.